root/components/SelectField.js

```diff
@@ -28,20 +28,22 @@ type Props = {|
   +field: FieldT<number | string>,
   +onChange?: (event: SyntheticEvent<HTMLSelectElement>) => void,
   +options: MaybeGroupedOptionsT,
+  +value?: number | string,
 |};
 
 const SelectField = ({
   allowEmpty = true,
   field,
   onChange,
   options,
+  value,
 }: Props) => (
   <select
     className="with-button"
     id={'id-' + field.html_name}
     name={field.html_name}
     onChange={onChange}
-    value={getSelectValue(field, options, allowEmpty)}
+    value={value || getSelectValue(field, options, allowEmpty)}
```

Comment:
There might be an issue here for blank options with `value=""`. Though I would prefer to keep all the "current" state inside `field` as mentioned in https://github.com/metabrainz/musicbrainz-server/pull/708#discussion_r212449001 and not allow any overrides.

Reply:
The following message can be dismissed by initializing text values to `''` in Perl:
> Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
Not sure it is the best way to address it though.